File: lingting-huawei/src/main/java/live/lingting/framework/huawei/iam/HuaweiIamTokenResponse.java
package live.lingting.framework.huawei.iam;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

/**
 * @author lingting 2024-09-13 11:52
 */
@Getter
@Setter
public class HuaweiIamTokenResponse {

	private Token token;

	public String getExpire() {
		return getToken().getExpire();
	}

	public String getIssued() {
		return getToken().getIssued();
	}

	@Getter
	@Setter
	public static class Token {

		@JsonProperty("expires_at")
		private String expire;

		@JsonProperty("issued_at")
		private String issued;

	}

}

File: lingting-huawei/src/main/kotlin/live/lingting/framework/huawei/iam/HuaweiIamTokenResponse.kt
package live.lingting.framework.huawei.iam

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @author lingting 2024-09-13 11:52
 */
class HuaweiIamTokenResponse {
    var token: Token = Token()

    val expire: String
        get() = token.expire

    val issued: String
        get() = token.issued

    class Token {
        @JsonProperty("expires_at")
        var expire: String = ""

        @JsonProperty("issued_at")
        var issued: String = ""
    }
}
